tasks/build.yml

```diff
@@ -0,0 +1,72 @@
+---
+
+- name: Create Prometheus Node Exporter directories
+  become: yes
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ prometheus_node_exporter.user }}"
+    group: "{{ prometheus_node_exporter.group }}"
+    mode: 0750
+  with_items:
+    - "{{ prometheus_node_exporter.path.install }}"
+    - "{{ prometheus_node_exporter.path.log }}"
+
+- name: Copy wrapper shell script
+  become: yes
+  copy:
+    src: prometheus_exporter.sh
+    dest: "{{ prometheus_node_exporter.path.install }}/prometheus_exporter.sh"
+    owner: "{{ prometheus_node_exporter.user }}"
+    group: "{{ prometheus_node_exporter.group }}"
+    mode: 0755
+
+- stat:
+    path: "{{ prometheus_node_exporter.download_dest }}/{{ prometheus_node_exporter.download_url.split('/')[-1] }}"
+  register: download_tar
+
+- name: Download Prometheus Node Exporter tarball
+  become: yes
+  get_url:
+    url: "{{ prometheus_node_exporter.download_url }}"
+    dest: "{{ prometheus_node_exporter.download_dest }}"
+    checksum: "{{ prometheus_node_exporter.checksum }}"
+  when: download_tar.stat.exists == False
+
+- name: Untar Prometheus Node Exporter tarball
+  become: yes
+  unarchive:
+    src: "{{ prometheus_node_exporter.download_dest }}/{{ prometheus_node_exporter.download_url.split('/')[-1] }}"
+    dest: "{{ prometheus_node_exporter.path.install }}"
+    owner: "{{ prometheus_node_exporter.user }}"
+    group: "{{ prometheus_node_exporter.user }}"
+    remote_src: True
+  when: download_tar.stat.exists == False
+
+- name: Ensure Prometheus Node Exporter directory does not exist
```

Comment:
You can probably fold this dir creation into the " Create Prometheus Node Exporter directories" task.